src/ut/alarm_trap_sender_test.cpp

```diff
@@ -0,0 +1,354 @@
+{
+public:
+  AlarmTrapSenderTest()
+  {
+    cwtest_completely_control_time();
+    cwtest_intercept_netsnmp(&_ms);
+  }
+
+  virtual ~AlarmTrapSenderTest()
+  {
+    delete _alarm_scheduler; _alarm_scheduler = NULL;
+    delete _alarm_table_defs; _alarm_table_defs = NULL;
+    _collector.call_all_callbacks();
+
+    cwtest_restore_netsnmp();
+    cwtest_reset_time();
+  }
+
+private:
+  std::set<NotificationType> _snmp_notifications;
+  MockNetSnmpInterface _ms;
+  CapturingTestLogger _log;
+  SNMPCallbackCollector _collector;
+  AlarmTableDefs* _alarm_table_defs;
+  AlarmScheduler* _alarm_scheduler;
+};
+
+// Safely set up an expect call for an SNMP trap send that will succeed.
+#define COLLECT_CALL(CALL) EXPECT_CALL(_ms, CALL).                                \
+  WillOnce(Invoke(&_collector, &SNMPCallbackCollector::collect_callback))
+#define COLLECT_CALLS(N, CALL) EXPECT_CALL(_ms, CALL).                            \
+  Times(N).                                                                       \
+  WillRepeatedly(Invoke(&_collector, &SNMPCallbackCollector::collect_callback))
+
+inline Matcher<netsnmp_variable_list*> EnterpriseTrapVars(oid trap_type,
+                                                          std::string MIB_version,
+                                                          std::string name,
+                                                          oid alarm_oid,
+                                                          oid resource_id,
+                                                          std::string severity,
+                                                          std::string description,
+                                                          std::string details,
+                                                          std::string cause,
+                                                          std::string effect,
+                                                          std::string action) {
+  return MakeMatcher(new EnterpriseTrapVarsMatcher(trap_type,
+                                                   MIB_version,
+                                                   name,
+                                                   alarm_oid,
+                                                   resource_id,
+                                                   severity,
+                                                   description,
+                                                   details,
+                                                   cause,
+                                                   effect,
+                                                   action));
+}
+
+inline Matcher<netsnmp_variable_list*> RFCTrapVars(RFCTrapVarsMatcher::TrapType trap_type,
+                                                   unsigned int alarm_index) {
+  return MakeMatcher(new RFCTrapVarsMatcher(trap_type, alarm_index));
+}
+
+TEST_F(AlarmTrapSenderTest, SetEnterpriseAlarm)
```

Comment:
Can you put a comment explaining what this test is covering (and for the other tests as well)